### ISSUE TITLE
Arrow Function Syntax Highlighting

### ISSUE THREAD

**Issue by paskozdilar** [typescript-language-server/typescript-language-server#824]
I opened an issue in NeoVim related to this issue: https://github.com/neovim/neovim/issues/26246

They have pointed out that I should report this issue to the server, not the client.

---

Here's a code example that reproduces the invalid syntax highlighing when used in NeoVim:

```typescript
function example(foo: string, bar: (baz: string, foobar: (foobaz: string, foobarbaz: () => void) => void) => void) {
  console.log('foo 1', foo);
  bar(foo, () => {
    console.log('foo 2', foo);
  });
}

function main() {
 example('foo', (baz: string, foobar: (foobaz: string, foobarbaz: () => void) => void) => {
   foobar(baz, () => {
      console.log('baz', baz);
   });
 });
}
```

The highlighting I get is this:

![image](https://github.com/typescript-language-server/typescript-language-server/assets/53006174/7082f164-c1c8-40ca-95b9-5eb02206bb6f)

The arrow function callback on the line 10 is not highlighted properly. In contrast, the arrow function callback on line 3 is highlighted properly.

**Comment by rchl** [comment]
It's not the server highlighting those. It's the syntax highlighting rules that your editor uses.
The comment in the neovim issue is a bit confusing since it mentions both.

**Comment by paskozdilar** [comment]
Okay, thanks for the quick feedback!

I was under the impression that language servers could (and did) override defualt syntax highlighting (especially in complex languages such as C++ and TypeScript), but I guess I was wrong :)

I have reported the issue to the appropriate place now.

**Comment by rchl** [comment]
server provides semantic highlighting but that just adds some highlights on top of normal syntax highlighting and is not responsible for what you see in that particular case
